app/src/services/dataService.js

```diff
@@ -269,5 +269,32 @@ module.exports = {
       }
     }
     return await this.getIPCPlan(ipcPlanId);
+  },
+
+  async update(business, contacts, ipcPlan, location, updatedBy) {
+    try {
+      await db.sequelize.transaction(async t => {
+        // currently, we are only allowed to update location data
+        // get the current data...
+        const currentLocation = await db.Location.findByPk(location.locationId);
+        // update the allowed fields...
+        currentLocation.startDate = location.startDate;
+        currentLocation.endDate =  location.endDate;
+        // set the user stamp...
+        currentLocation.updatedBy = updatedBy;
+        // save it...
+        await currentLocation.save({transaction: t});
+      });
+    } catch (e) {
+      log.error('dataService.update', e.message);
+      if (db.isRequiredFieldError(e)) {
+        throw new Problem(422, {detail: `Data cannot be updated. Required field is missing or empty: ${db.isRequiredFieldErrorMessage(e)}`});
+      } else if (db.isSyntaxError(e)) {
+        throw new Problem(422, {detail: `Data cannot be updated. Error with query: ${e.message}`});
```

Comment:
Minor, and see what the other guys think, but DB syntax is more of an internal server error isn't it?
If I'm PUTing a valid body to your API I wouldn't expect a 422 when the DB gets internally screwed up somehow for example.

Reply:
Now that you've mentioned this, it is a good point. Normally when I think of a 422, as it's a 4xx client error, that means that this code should reflect things that a user (browser) can fix - so if the client dispatches a missing or busted field, I'd expect a 422. A DB error on the other hand should be classified as a 5xx server error... didn't see anything we could really use though other than 500 Internal Server Error (and then say in detail what exactly happened).

Could always return HTTP 418 though if we wanted to be a teapot though :)